gateway/src/custom/policy/tls_client_certificate_authentication.lua

```diff
@@ -0,0 +1,83 @@
+local policy = require('apicast.policy')
+local _M = policy.new('Authenticate a client using his API Key or his SSL/TLS client certificate')
+
+local openssl = require('openssl')
```

Comment:
Depending on C library would be troublesome so the best approach is to use FFI like: https://github.com/SkyLothar/lua-resty-jwt/blob/master/lib/resty/evp.lua